AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests+Support.swift

```diff
@@ -0,0 +1,174 @@
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let note):
+                    result = .success(note)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                updateNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        })
+
+        wait(for: [updateNoteInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
+    }
+
+    func deleteNote(byId id: String,
+                    version: Int) -> Result<MutationSyncResult, GraphQLResponseError<MutationSyncResult>> {
+        let deleteNoteInvoked = expectation(description: "note was deleted")
+        var result: Result<MutationSyncResult, GraphQLResponseError<MutationSyncResult>>?
+        let request = GraphQLRequest<MutationSyncResult>.deleteMutation(modelName: SocialNote.modelName,
+                                                                        id: id,
+                                                                        version: version)
+        _ = Amplify.API.mutate(request: request, listener: { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let note):
+                    result = .success(note)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                deleteNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        })
+
+        wait(for: [deleteNoteInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
+    }
+
+    func queryNote(byId id: String) -> Result<MutationSyncResult?, GraphQLResponseError<MutationSyncResult?>> {
+        let queryNoteInvoked = expectation(description: "note was queried successfully")
+        var result: Result<MutationSyncResult?, GraphQLResponseError<MutationSyncResult?>>?
+        let request = GraphQLRequest<MutationSyncResult?>.query(modelName: SocialNote.modelName, byId: id)
+        _ = Amplify.API.query(request: request) { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let note):
+                    result = .success(note)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                queryNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        }
```

Comment:
Seems like there should be a way to consolidate this handler into a single function across all these functions?

Reply:
I couldn't find a way to do this, the problem is that return value is different for each case
createMutation -> Note
updateMutation -> Note
deleteMutation -> Note
query -> Note?
syncQuery -> SyncQueryResult